relayer/src/link/relay_path.rs

```diff
@@ -703,6 +711,18 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
 
         let msgs = odata.assemble_msgs(self)?;
 
+        telemetry!({
+            let (chain, counterparty, channel_id, port_id) = self.target_info(odata.target);
+
+            ibc_telemetry::global().tx_submitted(
```

Comment:
It seems like we're counting submitted transactions here, but this can lead to miscounting in two scenarios:

1. when there is an account sequence mismatch, we may submit the same transaction twice, so we can run into situations where we end up with
> tx_latency_confirmed_count{chain="ibc-1",channel="channel-0",counterparty="ibc-0",port="transfer"} 4
> tx_latency_submitted_count{chain="ibc-1",channel="channel-0",counterparty="ibc-0",port="transfer"} 5

and there will be a permanent offset between "submitted" and "confirmed", which is confusing: it can mean that a transaction was submitted but never confirmed, indicating some transactions maybe leaked


2. in the runtime, we may split this set of `msgs` across multiple transactions, but we only record a single record in the `tx_latency_submitted_count` metric. so the relayer may be submitting 10 separate transactions but only record 1 value here. I'm not sure this is a problem, it depends on users expectations actually.

Reply:
I wonder if this really matter? We are not counting transactions here but just measuring their latency, whether they're submitted twice or eventually rejected. The count here is only useful for computing the mean imho.

Reply:
This is a good point!

The question is if relayer operators use the `tx_latency_submitted_count` (resp. `tx_latency_submitted_count`) as a standalone value -- or do they use it solely for finding the latency mean? I will ask this Q in the next call with operators.


As a side note, while playing with, I think I found a way to fix the problem with a tentative solution here 45e3a94e

Reply:
The fix looks good, and makes the metric usable both ways so let's roll with that if someone can confirm it works. Won't have time to check myself unfortunately.